src/libmongoc/src/mongoc/mongoc-stream-tls-openssl.c

```diff
@@ -853,6 +815,81 @@ mongoc_stream_tls_openssl_new (mongoc_stream_t *base_stream, const char *host, m
    RETURN ((mongoc_stream_t *) tls);
 }
 
+/*
+ *--------------------------------------------------------------------------
+ *
+ * mongoc_stream_tls_openssl_new --
+ *
+ *       Creates a new mongoc_stream_tls_openssl_t to communicate with a remote
+ *       server using a TLS stream.
+ *
+ *       @base_stream should be a stream that will become owned by the
+ *       resulting tls stream. It will be used for raw I/O.
+ *
+ *       @trust_store_dir should be a path to the SSL cert db to use for
+ *       verifying trust of the remote server.
+ *
+ * Returns:
+ *       NULL on failure, otherwise a mongoc_stream_t.
+ *
+ * Side effects:
+ *       None.
+ *
+ *--------------------------------------------------------------------------
+ */
+
+mongoc_stream_t *
+mongoc_stream_tls_openssl_new (mongoc_stream_t *base_stream, const char *host, mongoc_ssl_opt_t *opt, int client)
+{
+   SSL_CTX *ssl_ctx = _mongoc_openssl_ctx_new (opt);
+
+   if (!ssl_ctx) {
+      RETURN (NULL);
+   }
+
+   if (!client) {
+      /* Only used by the Mock Server.
+       * Set a callback to get the SNI, if provided */
+      SSL_CTX_set_tlsext_servername_callback (ssl_ctx, _mongoc_stream_tls_openssl_sni);
+   }
+
+   return create_stream_with_ctx (base_stream, host, opt, client, ssl_ctx);
+}
+
+#if OPENSSL_VERSION_NUMBER >= 0x10100000L
+/*
+ *--------------------------------------------------------------------------
+ *
+ * mongoc_stream_tls_openssl_new_with_context --
+ *
+ *       Creates a new mongoc_stream_tls_openssl_t to communicate with a remote
+ *       server using a TLS stream, using an existing OpenSSL context.
+ *
+ *       Only called by mongoc_stream_tls_new_with_hostname_and_openssl_context.
+ *
+ *       @ssl_ctx is the global OpenSSL context for the mongoc_client_t
```

Comment:
```suggestion
 *       @ssl_ctx is the shared OpenSSL context for the mongoc_client_t
```

Suggest avoiding "global" since the shared OpenSSL context is not stored in a global variable.